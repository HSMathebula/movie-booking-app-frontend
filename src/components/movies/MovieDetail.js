import axios from 'axios';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router';
import { selectMovie } from '../../redux/movie/movie-detail';

function MovieDetail() {
  const movie = useSelector((state) => state.movie);
  console.log(movie);
  console.log('movie');
  const movietID = useParams();
  const dispatch = useDispatch();
  console.log(movietID);

  const fetchMovie = async () => {
    const response = await axios.get(`http://127.0.0.1:3000/api/v1/movies/${movietID}`).catch((err) => {
      console.log('Error', err);
    });
    dispatch(selectMovie(response));
  };

  useEffect(() => {
    fetchMovie();
  }, []);

  return (
    <div>MovieDetail</div>
  );
}

export default MovieDetail;
