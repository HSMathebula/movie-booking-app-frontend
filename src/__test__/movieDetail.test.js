import React from 'react';
import { BrowserRouter as Routers } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MovieDetail from '../components/movies/MovieDetail';

const movie = {
  description: 'this is a test movie',
  id: 1,
  title: 'Star wars',
  time: '1.20',
  duration: '20.00',
  picture:
    'https://image.shutterstock.com/image-vector/hawaii-beach-bar-retro-sign-600w-2156338567.jpg',
};

describe('testing the rendering of the movie details page', () => {
  it('', () => {
    const { container } = render(
      <Provider store={store}>
        <Routers>
          <MovieDetail movie={movie} />
        </Routers>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
