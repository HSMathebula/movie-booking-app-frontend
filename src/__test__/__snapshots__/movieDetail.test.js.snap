// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the rendering of the movie details page  1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="card"
    >
      <div
        class="container-fliud"
      >
        <div
          class="wrapper row"
        >
          <div
            class="preview col-md-6"
          >
            <div
              class="preview-pic tab-content"
            >
              <img
                class="img-fluid"
              />
            </div>
          </div>
          <div
            class="details col-md-6"
          >
            <h3
              class="product-title"
            />
            <div
              class="rating"
            >
              <div
                class="stars"
              >
                <span
                  class="fa fa-star checked"
                />
                <span
                  class="fa fa-star checked"
                />
                <span
                  class="fa fa-star checked"
                />
                <span
                  class="fa fa-star"
                />
                <span
                  class="fa fa-star"
                />
              </div>
              <h4
                class="review-no"
              >
                Time:
              </h4>
              <h4
                class="review-no"
              >
                Duration:
              </h4>
            </div>
            <p
              class="product-description"
            />
            <p
              class="vote"
            >
              <strong>
                91%
              </strong>
              of buyers enjoyed this Movie!
              <strong>
                (87 votes)
              </strong>
            </p>
            <div
              class="action"
            >
              <a
                href="/reserve_page"
              >
                <button
                  class="add-to-cart btn btn-default"
                  type="button"
                >
                  Reserve
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
